common/src/main/scala/com/pagerduty/scheduler/model/TaskKey.scala

```diff
@@ -40,18 +41,16 @@ case class TaskKey(
 }
 
 object TaskKey {
-  // NOTE:
-  // Since com.twitter.util.Time is always stored internally as UTC, this format will always be UTC
-  // as well, even if the time was created with a different zone.
-  val ScheduledTimeFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
-  private[scheduler] val TimeFormat = new TimeFormat(ScheduledTimeFormat)
+  val ScheduledTimeFormat = "yyyy-MM-dd'T'HH:mm:ss[.SSS]'Z'"
+  private[scheduler] val TimeFormat =
+    DateTimeFormatter.ofPattern(ScheduledTimeFormat).withZone(ZoneOffset.UTC)
```

Comment:
Any reason to not just use `DateTimeFormatter.ISO_INSTANT`? (which is [what Instant.parse uses](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#parse-java.lang.CharSequence-))